staging/src/k8s.io/apiserver/pkg/storage/etcd3/feature/feature_support_checker_test.go

```diff
@@ -0,0 +1,249 @@
+	tests := []struct {
+		testName              string
+		mockEndpointsVersion  []mockEndpointVersion
+		mockStatusReturnError error
+		expectedSupportResult bool
+		expectedError         error
+	}{
+		{
+			testName:              "Supported version, bound",
+			mockEndpointsVersion:  []mockEndpointVersion{{Version: "3.4.31", Endpoint: "localhost:2390"}},
+			expectedSupportResult: true,
+		},
+		{
+			testName: "Supported version, bound",
+			mockEndpointsVersion: []mockEndpointVersion{
+				{Version: "3.5.13", Endpoint: "localhost:2390"},
+				{Version: "3.5.14", Endpoint: "localhost:2391"}},
+			mockStatusReturnError: nil,
+			expectedSupportResult: true,
+			expectedError:         nil,
+		},
+		{
+			testName:              "Supported version",
+			mockEndpointsVersion:  []mockEndpointVersion{{Version: "3.6.0", Endpoint: "localhost:2390"}},
+			expectedSupportResult: true,
+		},
+		{
+			testName:              "Unsupported version",
+			mockEndpointsVersion:  []mockEndpointVersion{{Version: "3.4.30", Endpoint: "localhost:2390"}},
+			expectedSupportResult: false,
+		},
+		{
+			testName:              "Unsupported version",
+			mockEndpointsVersion:  []mockEndpointVersion{{Version: "3.5.0", Endpoint: "localhost:2390"}},
+			expectedSupportResult: false,
+		},
+		{
+			testName:              "Unsupported version",
+			mockEndpointsVersion:  []mockEndpointVersion{{Version: "3.5.12", Endpoint: "localhost:2390"}},
+			expectedSupportResult: false,
+		},
+		{
+			testName:              "Mock client status return error",
+			mockEndpointsVersion:  []mockEndpointVersion{{Version: "3.5.13", Endpoint: "localhost:2390"}},
+			mockStatusReturnError: fmt.Errorf("Error from status"),
+			expectedSupportResult: false,
+			expectedError:         fmt.Errorf("failed checking etcd version, endpoint: %q: %w", "localhost:2390", errors.New("Error from status")),
+		},
+		{
+			testName:              "Malformed version, error of malformed version logged via klog",
+			mockEndpointsVersion:  []mockEndpointVersion{{Version: "3.5.--a", Endpoint: "localhost:2390"}},
+			expectedSupportResult: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			// Mock Etcd client
+			mockClient := &MockEtcdClient{EndpointVersion: tt.mockEndpointsVersion, Error: tt.mockStatusReturnError}
+
+			ctx := context.Background()
+			supported, err := endpointSupportsRequestWatchProgress(ctx, mockClient, "localhost:2390")
+
+			// Assertions
+			assert.Equal(t, tt.expectedSupportResult, supported)
+			assert.Equal(t, tt.expectedError, err)
+		})
+	}
+}
+
+func TestSupports(t *testing.T) {
+	tests := []struct {
+		testName                      string
+		featureName                   string
+		mockEndpointsVersion          []mockEndpointVersion
+		mockSecondaryEndpointsVersion []mockEndpointVersion
+		expectedResult                bool
+		expectedSecondaryResult       bool
+		expectedError                 error
+		expectedCheckClientError      error
+	}{
```

Comment:
```suggestion
	type testCase struct {
		endpointsVersion []mockEndpointVersion
		expectedResult     bool
		expectedError       error
	}
	tests := []struct {
		testName                      string
		firstRound *testCase
		secondRound *testCase
	}{
```

Reply:
No need to test other `featureName` values in the same field. You can have a separate tests, `TestSupportsRequestWatchProgress` and `TestSupports`